### PR TITLE
Configure Sphinx docs and Travis-CI deployment to GitHub Pages

### PR DESCRIPTION
**Pull request by gnidan**
This PR has two parts:
- Sphinx docs set-up
- Deployment

Sphinx
-------

[Sphinx](http://www.sphinx-doc.org/en/master/) is an extensible docs generator. This PR enables extensions for:
- JSON schema support via [sphinx-jsonschema](http://sphinx-jsonschema.readthedocs.io/en/latest/index.html)
- Support for [PlantUML](http://plantuml.com/) diagrams via [sphinxcontrib-plantuml](https://github.com/sphinx-contrib/plantuml/)
- Toggle-able dropdowns via [sphinxcontrib-contentui](https://github.com/ulrobix/sphinxcontrib-contentui)

Deployment
-------------

This PR sets up Travis deployment to gh-pages so that anyone who opens a PR can preview their changes.  **For instance:** [Preview this PR](http://ethpm.github.io/ethpm-spec/contrib/ethpm/gnidan/sphinx)!

Included script `./bin/deploy` checks out the `gh-pages` branch and copies the built docs output into the directory structure on that branch:
- Builds directly on `master` copy to branch root, to resolve at **ethpm.github.io/ethpm-spec/**. 
  _(Note: this link won't work until this PR is merged)_
- Commits on other branches and PRs are placed in `./contrib/<remote-fork-name>/<branch-name>`, to resolve at **ethpm.github.io/contrib/`<remote-fork-name>`/`<branch-name>`**.

To enable this, I have create a single-purpose deploy key with repository write-access, encrypted it with Travis's public key, and configured the CI build to hook everything up. For reference, I've adapted a process described in [this gist](https://gist.github.com/domenic/ec8b0fc8ab45f39403dd).

Other Notes
------------

- Plug in `validate.py` into the Travis-CI build process, since that was previously not automated.
- This PR doesn't actually contain docs yet, to keep this work isolated.

### CODE REVIEW

**Comment by pipermerriam** [comment]
I'd highly recommend using read-the-docs over gh-pages for hosting.  They have solid facilities for building and hosting the docs, including multiple versions across multiple releases, etc.

**Comment by gnidan** [comment]
@pipermerriam I would prefer RTD also, but I have two concerns/questions:
- It'd be nice to support previews of any contributors changes before they are merged.
- Can RTD run plantuml.jar for us?

It's been awhile since I've done research on the matter, but last I checked, this was lacking. Please advise!

**Comment by pipermerriam** [comment]
It just support custom themes, extensions, and in general, custom builds.  dunno about `plantuml.jar` but I suspect it's doable.  What specifically is that for?

**Comment by gnidan** [comment]
PlantUML is just there in case diagrams would be useful. It's certainly not a hard requirement. I just like diagrams. 🙃 

I will research what the support is like with RTD these days. Thanks!

**Comment by pipermerriam** [comment]
Also found this with some quick searching. https://plantweb.readthedocs.io/

**Comment by gnidan** [comment]
> Also found this with some quick searching. https://plantweb.readthedocs.io/

I'm aware of this tool, but it requires a PlantUML server. There's a [public server](https://plantweb.readthedocs.io/#plantuml-server), but I don't know about reliability, or any performance impact.

Anyway, it looks like there's an open issue https://github.com/rtfd/readthedocs.org/issues/407 for PlantUML support, but [this RTD site](http://build-me-the-docs-please.readthedocs.io/en/latest/Using_Sphinx/UsingGraphicsAndDiagramsInSphinx.html#using-plantuml) seems to have it working (and from the looks of it, with the JAR self-hosted somehow? Unclear at first glance.)

**Comment by gnidan** [comment]
@pipermerriam Do you feel strongly **against** GitHub Pages, or just strongly **in favor** of RTD?

If it's the latter, I propose this as an initial solution, since it works and is arguably not too complex; we can always switch when it becomes clear that this project needs a more mature solution like RTD.

**Comment by gnidan** [comment]
For the other concern (the pre-merge previews), it seems like this is a requested feature, not yet available: https://github.com/rtfd/readthedocs.org/issues/2465

**Comment by pipermerriam** [comment]
@gnidan I'll defer to you.  All that really matters is that there is reliably hosted documentation that is up-to-date.

**Comment by gnidan** [comment]
Thanks @pipermerriam. I am confident that this will suffice – Truffle's been using a similar GH pages build for almost a year without major problems.

Mind reviewing the implementation?

Also, anyone else have thoughts?

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+
+# -- Project information -----------------------------------------------------
+
+project = u'EthPM Package Manifest Specification'
+copyright = u'2018, Piper Merriam, et al.'
+author = u'Piper Merriam, et al.'
+
+# The short X.Y version
+version = u''
```
Comment (pipermerriam) [inline docs/source/conf.py]: This is python2 unicode string syntax. We should probably be using python3 which doesn't have the `u` prefix.

**Comment by gnidan** [comment]
Re: build error: I forgot to include .gitkeep for `_static` and `_templates` because they're empty right now. Fix that and the string syntax.

@njgheorghita We should probably wait to add this to the README, since the docs are totally empty. I figure, we can set up the link once we moved the `package.spec.md` into Sphinx.

I'll merge this tomorrow assuming no objections. Thank you very much!

**Pull request merged**